File: LeetCode/array/Difference array/p1109/java/solution.java
class Solution {
    int[] diff;
    public int[] corpFlightBookings(int[][] bookings, int n) {
        diff = new int[n];
        for(int i=0;i<bookings.length;i++) {
            diff[bookings[i][0]-1] += bookings[i][2];
            if(bookings[i][1]<n) {
                diff[bookings[i][1]] -= bookings[i][2];
            }
        }
        for(int i=1;i<n;i++){
            diff[i] += diff[i-1];
        }
        return diff;
    }
}

File: LeetCode/array/Difference array/p1109/c++/solution.cpp
class Solution {
public:
    vector<int> corpFlightBookings(vector<vector<int>>& bookings, int n) {
        vector<int> ans(n,0);
        for(auto &N:bookings){
            ans[N[0]-1] += N[2];
            if(N[1]<n){
                ans[N[1]] -= N[2];
            }
        }
        for(int i=1;i<n;i++){
            ans[i]+=ans[i-1];
        }
        return ans;
    }
};